common/src/main/java/me/shedaniel/clothconfig2/impl/builders/FieldBuilder.java

```diff
@@ -72,8 +77,59 @@ public boolean isRequireRestart() {
         return requireRestart;
     }
     
+    public B setErrorSupplier(@Nullable Function<T, Optional<Component>> errorSupplier) {
+        this.errorSupplier = errorSupplier;
+        return (B) this;
+    }
+    
+    @Deprecated
+    @ApiStatus.ScheduledForRemoval
     public void requireRestart(boolean requireRestart) {
         this.requireRestart = requireRestart;
     }
     
+    public B requiresRestart(boolean requireRestart) {
+        this.requireRestart = requireRestart;
+        return (B) this;
+    }
+    
+    public B requiresRestart() {
+        this.requireRestart = true;
+        return (B) this;
+    }
+    
+    public B setSaveConsumer(Consumer<T> saveConsumer) {
+        this.saveConsumer = saveConsumer;
+        return (B) this;
+    }
+    
+    public B setDefaultValue(Supplier<T> defaultValue) {
+        this.defaultValue = defaultValue;
+        return (B) this;
+    }
+    
+    public B setDefaultValue(T defaultValue) {
+        this.defaultValue = () -> defaultValue;
+        return (B) this;
+    }
+    
+    public B setTooltip(Supplier<Optional<Component[]>> tooltipSupplier) {
+        this.tooltipSupplier = str -> tooltipSupplier.get();
```

Comment:
I would recommend renaming this internal variable to `tooltipGetter` or something other than `Supplier`, since it isn't a `Supplier`.

You may be interested in having a look at the refactor I undertook in 17ce11f2a515d14aff1cfd444861af7310ecdb75, though it's sort of orthogonal to your work so far in this PR